prody/ensemble/functions.py

```diff
@@ -364,18 +364,40 @@ def buildPDBEnsemble(refpdb, PDBs, title='Unknown', labels=None, seqid=94, cover
     :arg unmapped: A list of PDB IDs that cannot be included in the ensemble. This is an 
     output argument. 
     :type unmapped: list
+
+    :arg alignment_list: list of alignment duplets for mapping function
+        default is list of None
+    :type alignment_list: list
+
+    :arg alignments_list: list of alignments for mapping function
+        default is list of None
+    :type alignments_list: list
     """
 
     if not isinstance(refpdb, (Chain, Segment, Selection, AtomGroup)):
         raise TypeError('Refpdb must be a Chain, Segment, Selection, or AtomGroup.')
     
     if labels is not None:
         if len(labels) != len(PDBs):
-            raise TypeError('Labels and PDBs must have the same lengths.')
+            raise TypeError('labels and PDBs must have the same lengths.')
+
+    alignment_list = kwargs.get('alignment_list', None)
+    if alignment_list is None:
+        alignment_list = [None]* len(PDBs)
+    else:
+        if len(alignment_list) != len(PDBs):
+            raise TypeError('alignment_list and PDBs must have the same lengths.')
+
+    alignments_list = kwargs.get('alignments_list', None)
+    if alignments_list is None:
+        alignments_list = [None]* len(PDBs)
+    else:
+        if len(alignments_list) != len(PDBs):
+            raise TypeError('alignments_list and PDBs must have the same lengths.')
```

Comment:
alignment_list is wrapped in kwargs so it does not need to be processed. See #537

Reply:
ok

Reply:
It still needs documenting though I guess

Reply:
yes sure